app/controllers/customers_controller.rb

```diff
@@ -1,12 +1,12 @@
-class CustomerController < ApplicationController
+class CustomersController < ApplicationController
   before_action :authenticate_user!
   
   def new
     @customer = User.new
   end
 
   def create
-   customer = User.new(customer_params)
+    customer = User.new(customer_params)
     customer.password = Settings.user.password
     customer.add_role :customer
```

Comment:
@RaviMariya-BTC Adding role as customer, You can put inside the model as the callback. But make sure that this method should call only when the customer register with application.